report/main.tex

```diff
@@ -0,0 +1,810 @@
+The supervisor node is a module designed to manage high level decisions and keep track of useful race metrics. It provides an interface for signal detection nodes to indicate that an event has occurred. Using it's knowledge of the current state of the robot, it decides when to start and stop the car. To start and stop the car it coordinates with a twist multiplexer node to prioritize which twist messages get published to the robot. The twist multiplexer is a third party node  that takes various twist message inputs, and based on the priority of the messages and any locks, will pick one to output to the ROS topic that the vehicle controller subscribes to. Currently, there are three sources of twist messages: joystick, path planner, and the supervisor node itself. The supervisor node's command topic is simply used to sending null vectors to stop the robot. Additionally, the supervisor node publishes a Boolean to two topics that the twist multiplexer looks for to set priority locks on the incoming twist messages. One topic is used to lock out all messages except for the supervisor’s topic, which is used to bring the robot to a complete stop. The other topic is to remove this lock, allowing all messages through, at which point the message with the highest priority will go through. The twist message priority is as follows: supervisor, joystick, path planner.
+The supervisor node also keeps track of some useful variables; namely the race type, start and end times, lap count, average speed, and battery life. When the race is finished, a few of these variables are stored and saved into a text file. These variables are used to make important decisions regarding when to start and stop the race. The race is started when a traffic light signal is received, and is ended when either the lap count is reached for that race (1 for drag, 3 for circuit), which is incremented by an end line detection signal, or if the battery life dips below 10%.
+
+\subsection{Computer Vision}
+This competition year huge strides we're made in improving the robustness and functionality of our computer vision software. To improve the robustness of our lane detection system a shadow removal algorithm was added as a pre-processing step on each of our cameras. To improve our traffic light detection in all conditions a machine learning classifier was added to help detect traffic lights in images. End line line detection was improved considerable over past years as well largely due to the addition of a new hysteresis filter on the camera stream. 
+
+\subsubsection{Shadow Removal}
+In last year’s competition, the robot had difficulty in reliably tracking lane lines under variable lighting conditions, such as when buildings or robots cast shadows onto the track. As a result, this year’s team implemented a shadow detection and removal pre-processing step, based on work done by Murali et al[cite], in order to address this issue.
```

Comment:
`last year’s competition` is using the wrong type of quotation mark, it doesn't show up in the pdf. This is a problem with all the quotation marks in my section, maybe a result of copy-pasting from Word.

Don't forget to add the citation for `Murali et al[cite]`, and there should be a period and space after the abbreviation `al`, like `et al. [cite]`.